.github/workflows/java-ci.yml

```diff
@@ -28,12 +28,6 @@ on:
     - '.github/workflows/cancel-duplicate-workflow-runs.yml'
     - '.gitignore'
     - 'dev/**'
-    - 'mr/**'
-    - 'hive3/**'
-    - 'hive3-orc-bundle/**'
-    - 'hive-runtime/**'
-    - 'spark/**'
-    - 'flink/**'
```

Comment:
In my view,  the overall checkstyle check is necessary.  I think the problem is:  removing all those ignored engine paths will trigger the extra `iceberg-core` unit tests check. Because if any engine related changes happen , it will always run the iceberg-core unit tests  (even we don't ) I think.

Reply:
I agree with @openinx. This common rule is the one that should be running the checks. This is to avoid making all of the CI builds more expensive because checkstyle and errorprone take a long time to run.

One thing that we still need to do after this is enable all Flink, Spark, and Hive versions in the job config, not just the default ones.